### ISSUE TITLE
make: *** [Makefile:604: br-rootfs-pack] Error 2 Error: Build board milkv-duo failed!

### ISSUE THREAD

**Issue by muditmadan** [milkv-duo/duo-buildroot-sdk#62]
I keep getting this error when I try to compile using wsl 2

make: *** [Makefile:604: br-rootfs-pack] Error 2
Error: Build board milkv-duo failed!

**Comment by carbonfix** [comment]
You can clone a copy of the code again to test whether the compilation is normal. If it is normal, it means that your old project needs to be cleaned and recompiled.

**Comment by xpww** [comment]
I also have the same problem, have you solved it?

**Comment by xpww** [comment]
问题已经解决，是因为conda环境的问题，可以conda install conda-forge::expect或者conda deactivate后再执行脚本

**Comment by jiapei100** [comment]
> 问题已经解决，是因为conda环境的问题，可以conda install conda-forge::expect或者conda deactivate后再执行脚本

Nope...  我没有安装 conda ....

**Comment by thirstyone** [comment]
same error. here's the end of br.log: 
```
2024-02-13T16:32:09 patching file test/aclocal.m4
2024-02-13T16:32:09 patching file test/configure
2024-02-13T16:32:09 patching file test/tclock.c
2024-02-13T16:32:09 
2024-02-13T16:32:09 Applying 0001-gcc-5.x-MKlib_gen.patch using patch:
2024-02-13T16:32:09 patching file ncurses/base/MKlib_gen.sh
2024-02-13T16:32:09 Hunk #1 succeeded at 501 (offset -4 lines).
2024-02-13T16:32:09 >>> host-ncurses 6.1 Updating config.sub and config.guess
2024-02-13T16:32:09 for file in config.guess config.sub; do for i in $(find /duo-buildroot-sdk-Duo-V1.0.8/buildroot-2021.05/output/milkv-duo256m-lite_musl_riscv64/build/host-ncurses-6.1 -name $file); do cp support/gnuconfig/$file $i; done; done
2024-02-13T16:32:10 >>> host-ncurses 6.1 Patching libtool
2024-02-13T16:32:10 make[1]: Leaving directory '/duo-buildroot-sdk-Duo-V1.0.8/buildroot-2021.05'

```

**Comment by Plaque-fcc** [comment]
> You can clone a copy of the code again to test whether the compilation is normal. If it is normal, it means that your old project needs to be cleaned and recompiled.

Yeah, cloning (and downloading everything again) fixes the problem.

For me, this happened tonight, and last time I build a firmware was v1.0.7 somewhere.

**Comment by Freireg** [comment]
I'm having the same issue with V1.0.8 and V1.0.9 while attempting to build from a WSL ubuntu 22. And if I try to build directly from buildroot I receive the following: 
```
➜  duo-buildroot-sdk git:(Duo-V1.0.8) defconfig cv1800b_milkv_duo_sd
 Run  function
_call_kconfig_script:11: no such file or directory: /home/freire/duo-buildroot-sdk/build/scripts/.py
```
Any suggestions on how to solve it?

## Update:
In a ubuntu VM it seems to be working just as intended

**Comment by micropythondevel** [comment]
br-rootfs-pack sigue fallando

**Comment by xpww** [comment]
In the end, I had no choice, so I used the docker image to compile

**Comment by Bender1983** [comment]
there is also another post to this problem, see https://github.com/milkv-duo/duo-buildroot-sdk/issues/74

**Comment by MarkMLl** [comment]
I've been chewing at this for a few days, and there appear to be three distinct problems.

Context: host is Debian 12 "Bookworm", build environment ditto running under UML.

The first is that python-lxml will not build, so as an interim measure remove it from package/Config.in (it can be left in package/Config.in.host).

The second is that the final xz fails. Trimming the path from the command line and running manually it boils down to

<pre>
$ xz -9 -C crc32 -c -T 2 ~/duo-buildroot-sdk-develop/buildroot-2021.05/output/milkv-duo_musl_riscv64/images/rootfs.tar > ~/duo-buildroot-sdk-develop/buildroot-2021.05/output/milkv-duo_musl_riscv64/images/rootfs.tar.xz
Killed
</pre>

Removing the -9 option fixes it, replacing that with -z 9 or --compress 9 results in an error which suggests that the compression option is problematic in at least some variants of xz.

The root cause appears to be here:

<pre>
buildroot-2021.05/fs/common.mk

ifeq ($$(BR2_TARGET_ROOTFS_$(2)_XZ),y)
ROOTFS_$(2)_DEPENDENCIES += host-xz
ROOTFS_$(2)_COMPRESS_EXT = .xz
ROOTFS_$(2)_COMPRESS_CMD = xz -9 -C crc32 -c
</pre>

With those two changes, buildroot should complete but there's still a further problem in genimage:

<pre>
...
INFO: cmd: "mkdir -p "/home/markMLl/duo-buildroot-sdk-develop/install/soc_cv1800b_milkv_duo_sd"" (stderr):
INFO: vfat(boot.vfat): cmd: "dd if=/dev/zero of="/home/markMLl/duo-buildroot-sdk-develop/install/soc_cv1800b_milkv_duo_sd/boot.vfat" seek=134217728 count=0 bs=1 2>/dev/null" (stderr):
INFO: vfat(boot.vfat): cmd: "mkdosfs  -n 'boot' '/home/markMLl/duo-buildroot-sdk-develop/install/soc_cv1800b_milkv_duo_sd/boot.vfat'" (stderr):
sh: 1: mkdosfs: not found
INFO: vfat(boot.vfat): cmd: "rm -f "/home/markMLl/duo-buildroot-sdk-develop/install/soc_cv1800b_milkv_duo_sd/boot.vfat"" (stderr):
ERROR: vfat(boot.vfat): failed to generate boot.vfat
gnimage for milkv-duo failed!
</pre>

In device/gen_burn_image_sd.sh add explicit paths to host utilities:

<pre>
...
[ -d tmp ] && rm -rf tmp

genimage --mkdosfs /usr/sbin/mkdosfs --tune2fs /usr/sbin/tune2fs --e2fsck /usr/sbin/e2fsck --config ${TOP_DIR}/device/${MV_BOARD}/genimage.cfg --rootpath fs/ --inputpath ${PWD} --outputpath ${PWD}
if [ $? -eq 0 ]; then
    echo "gnimage for ${MV_BOARD} success!"
else
    echo "gnimage for ${MV_BOARD} failed!"
fi
</pre>

I believe that results in a successful build, but will re-test.

I've not yet looked at the Docker image, but suspect that it's using a different or patched version of BuildRoot.

**Comment by MarkMLl** [comment]
So assuming v1.0.9 of duo-buildroot-sdk-develop:

In buildroot-2021.05/package/Config.in remove the line

`"package/python-lxml/Config.in"`

In buildroot-2021.05/fs/common.mk remove the -9 option from

`ROOTFS_$(2)_COMPRESS_CMD = xz -9 -C crc32 -c`

In device/gen_burn_image_sd.sh add host paths:

<pre>
genimage --mkdosfs /usr/sbin/mkdosfs --tune2fs /usr/sbin/tune2fs --e2fsck /usr/sbin/e2fsck \
    --config ${TOP_DIR}/device/${MV_BOARD}/genimage.cfg --rootpath fs/ --inputpath ${PWD} --outputpath ${PWD}
</pre>

Then build using ./build.sh milkv-duo as documented, which eventually produces a rewarding

`Create SD image successful: out/milkv-duo-20240409-1235.img`

...which I can confirm boots and runs.

While the unimplemented option to xz appears fairly uncontroversial, a proper fix for the python-lxml package and incorporating the correct host paths into device/gen_burn_image_sd.sh are "left as an exercise".
